source/common/http/http1/conn_pool.cc

```diff
@@ -96,12 +96,17 @@ ConnectionPool::Cancellable* ConnPoolImpl::newStream(StreamDecoder& response_dec
   if (host_->cluster().resourceManager(priority_).pendingRequests().canCreate()) {
     bool can_create_connection =
         host_->cluster().resourceManager(priority_).connections().canCreate();
+    bool can_create_pending_connection =
+        host_->cluster().resourceManager(priority_).pendingConnections().canCreate();
     if (!can_create_connection) {
       host_->cluster().stats().upstream_cx_overflow_.inc();
     }
+    if (!can_create_pending_connection) {
+      host_->cluster().stats().upstream_cx_pending_overflow_.inc();
+    }
 
     // If we have no connections at all, make one no matter what so we don't starve.
-    if ((ready_clients_.size() == 0 && busy_clients_.size() == 0) || can_create_connection) {
+    if ((ready_clients_.size() == 0 && busy_clients_.size() == 0) || (can_create_connection && can_create_pending_connection)) {
```

Comment:
Should `can_create_pending_connection` be tested and stats published only when `(ready_clients_.size() == 0 && busy_clients_.size() == 0) || can_create_connection` evaluates to `true`? If not, we'd have both `upstream_cx_overflow_` and `upstream_cx_pending_overflow_` incremented while only one of them caused the circuit to break.